vermouth/processors/annotate_mut_mod.py

```diff
@@ -105,22 +105,22 @@ def residue_matches(resspec, residue_graph, res_idx):
     res_node = residue_graph.nodes[res_idx]
     residue = {key: res_node.get(key)
                for key in 'chain resid resname insertion_code'.split()}
-    out = True
-    if resspec.get('resname', '')[-3:] == 'ter':
-        # Find all residues with degree 1: the ones with the lowest resid will
-        # be cter, the ones with the highest resid nter.
-        termini = [idx for idx in residue_graph if residue_graph.degree[idx] == 1]
-        get_resid = lambda idx: residue_graph.nodes[idx].get('resid')
+    if resspec.get('resname', '')[-3:] == 'ter' and residue_graph.degree[res_idx] == 1:
```

Comment:
Suppose I name a modification random-string-ter, the code would treat it as C or N terminus. This is something which becomes very hard to debug for the user and potentially us.  So I'd change the check to check for the complete name.

Reply:
This gets caught in 121

Reply:
I know but I am still not convinced that this is a good choice, because fundamentally you are treating C and N terminus special and the code is also not general for any terminus that would get caught in this. It is only for proteins specifically following the standard convention of writing N to C terminus. So I'd be very much in favor of either making it completely special checking c-ter or n-ter or making it general. But as you write in the fixme the general solution will probably require DiGraphs.